### DISCUSSION TITLE
Zweite Woche

### DISCUSSION THREAD

**Discussion by PetrusQn** [MelonigeMelone/xCalendar#6] [Blogkanal]
## 2 Angular 

Wie angekündigt haben wir in den letzten Tagen ein Angular Projekt erstellt, in welchem wir die Webanwendung für xCalendar programmieren möchten. Als Entwicklungsumgebung haben wir uns für _Visual Studio Code_ entschieden. Hierfür musste neben dem _Angular Command-Line-Interface_ auch _Node.js_ installiert werden.   

## Jenkins

Wir haben einen Jenkins aufgesetzt und Basis Einstellungen vorgenommen eine direkte Verknüpfung mit dem Repository und der Integration in den fertigen Workflow fehlt noch.
Das Einrichten war recht einfach über einen bereits vorhanden Debian RootServer wurde das Jenkins Package installiert und die Basis Konfiguration vorgenommen. Zudem wurde er auf eine bereits bestehende Domain weitergeleitet über eine Reverse Proxy Konfiguration.

## Datenbank

Wir haben eine Datenbank sowie einen Nutzer mit vollen rechten auf einer MYSQL Datenbank eingerichtet. Zudem ein entsprechende Webpanel (phpmyadmin) zum einsehen und bearbeiten der Daten aufgesetzt.Hier mussten lediglich Befehle in der MySQL Konsole auf dem Linux Rootserver ausgeführt werden, da die Datenbank bereit im Vorfeld existierte. 

## SRS
Die SRS mit den Informationen und den weiteren Aufgaben der Woche findet ihr hier: https://github.com/MelonigeMelone/xCalendar/discussions/7

## Sprint Überblick
Unser Sprint ist fast zu Ende und es sind nahezu alle Aufgaben erledigt, die restlichen Aufgaben werden in Software Engineering oder vor dem Wochenende noch erledigt

**Comment by Christian-2003** [comment]
Hallo zusammen,

eurem Blogeintrag entnehme ich, dass ihr diese Woche ziemlich viel geschafft habt. Könntet ihr nächste Woche theoretisch direkt mit dem Programmieren beginnen? Das würde dann ja mit dem Beginn des nächsten Sprints ziemlich gut zusammenpassen.

Was mich interessieren würde ist, was genau noch bei der Verknüpfung von Jenkins mit eurem Repository fehlt und wie ihr das in euren Workflow integrieren wollt.

Laut eurem SRS scheint ihr schon einen ziemlich genauen Plan für die Implementierung zu haben, zumindest lässt das vorhandene UML-Klassendiagramm darauf schließen. Habt ihr diesbezüglich auch schon irgendwas programmiert?

Bezüglich eures SRS würde mich ein UI-Mockup für euren Kalender interessieren, ich kann mir diesbezüglich nämlich nichts vorstellen, bin aber auch nicht so gut darin. 😅

Aber insgesamt war das ein aufschlussreicher und gelungener Blogeintrag, weiter so.

Ich freue mich schon darauf, nächste Woche weiterzulesen.
